### PR TITLE
Moved common objects from node agent to commons repo

### PR DESCRIPTION
**Pull request by nnDarshan**
tendrl-bug-id: Tendrl/commons#308
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@r0h4n @shtripat @anmolbabu @nthomas-redhat Please review

File: tendrl/commons/objects/cluster_node_context/__init__.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+LOG = logging.getLogger(__name__)
+
+
+class ClusterNodeContext(objects.BaseObject):
+
+    def __init__(self, machine_id=None, node_id=None, fqdn=None,
+                 tags=None, status=None, *args, **kwargs):
+        super(ClusterNodeContext, self).__init__(*args, **kwargs)
+        self.value = 'clusters/%s/nodes/%s/NodeContext'
```
Comment (r0h4n) [inline tendrl/commons/objects/cluster_node_context/__init__.py]: Apart "value" you can get every other attribute directly from NS.node_context.$attribute

You wont need any of the below functions too apart from render()

Comment (nnDarshan) [inline tendrl/commons/objects/cluster_node_context/__init__.py]: Done

File: tendrl/commons/objects/cluster_node_context/__init__.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        self.machine_id = machine_id or NS.node_context.machine_id
+        self.node_id = node_id or NS.node_context.node_id
+        self.fqdn = fqdn or NS.node_context.fqdn
+        self.tags = tags or NS.config.data['tags']
```
Comment (r0h4n) [inline tendrl/commons/objects/cluster_node_context/__init__.py]: Please use json.loads(node_context.tags)

File: tendrl/commons/objects/cluster_node_context/__init__.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+LOG = logging.getLogger(__name__)
+
+
+class ClusterNodeContext(objects.BaseObject):
+
+    def __init__(self, machine_id=None, node_id=None, fqdn=None,
+                 tags=None, status=None, *args, **kwargs):
+        super(ClusterNodeContext, self).__init__(*args, **kwargs)
+        self.value = 'clusters/%s/nodes/%s/NodeContext'
+        self.machine_id = machine_id or NS.node_context.machine_id
```
Comment (r0h4n) [inline tendrl/commons/objects/cluster_node_context/__init__.py]: Do a "node_context = NS.node_context.load()"

the use "node_context.machine_id" and other attributes

File: tendrl/commons/objects/cluster_node_context/__init__.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        self.node_id = node_id or NS.node_context.node_id
+        self.fqdn = fqdn or NS.node_context.fqdn
+        self.tags = tags or NS.config.data['tags']
+        self.status = status or "UP"
```
Comment (r0h4n) [inline tendrl/commons/objects/cluster_node_context/__init__.py]: node_context.status

**Pull request merged**